model/shared/src/main/scala/explore/model/ConstraintsSummary.scala

```diff
@@ -24,7 +28,8 @@ final case class ConstraintsSummary(
   imageQuality:    ImageQuality,
   cloudExtinction: CloudExtinction,
   skyBackground:   SkyBackground,
-  waterVapor:      WaterVapor
+  waterVapor:      WaterVapor,
+  obsCount:        NonNegInt
```

Comment:
I wonder if this should be an `Option` and only read it in the obs view

Reply:
🤔 It definitely isn't needed everywhere. On the other hand, do we want to deal with the optionality in the places it is required?

Reply:
Yeah that's the question, and definitely I don't want to have different classes for different queries

Reply:
I guess we could do `getOrElse(0)` where it is needed. It's a little loss of safety, but in practice it isn't likely to be a problem.

Reply:
Why do we need the count in `ConstraintsSummary`? I thought `ConstraintsSummary` was only being used to build the badges. I would expect it in `ConstraintSetModel` and maybe the logic to display warning within `ConstraintsPanel`, which will be shared in both the Constraints and Observation views.

Reply:
Answering my own question: we probably need it in the table to show when no item is selected in the constraints view. In that case, we do need it in `ConstraintsSummary` if that's what we are using there. In that case, we should probably use the same live query to feed data to both the tree and the table.

Reply:
I want to reuse the constraint editor on the obs view (like we do for targets ) and that takes a ConstraintsSummary, I just want to get the obs count there to implement the custom logic for shared/not shared constraint sets

Reply:
I must be missing something. Doesn't the constraint editor (`ConstraintsPanel`) take a `ConstraintSetModel`?

Reply:
You're right. we only pass the constraints to query a `ConstrantSetModel` and use it in `ConstraintsPanel`

I'm thinking of showing if you the constraint is shared on the tile, thus I need it on the Obs query.
u

Reply:
I'm going to try making `ConstraintsSummary` a trait with the 4 enums and the `summayString` method and let clue generate types to store results but force them to extend such trait. This should be in line with the approach of not decoding into our model and instead converting (or in this case extending), which would in turn remove the need for our own decoder.

Reply:
Ok let's close this